setup.cfg

```diff
@@ -37,6 +37,7 @@ install_requires =
     responses>=0.13.0
     Jinja2>=2.10.1
     importlib_metadata ; python_version < '3.8'
+    typing_extensions
```

Comment:
Having to introduce this is unfortunate, but unavoidable. I will probably only merge this PR as part of a minor release (4.3.0), to reduce the impact.  (I.e., reduce the number of users that do not update their dependencies along with Moto, and then complain about missing imports..)

Reply:
I think we can avoid it as a runtime dependency, by gating it behind `if TYPE_CHECKING:`, if you think that's better.

On the other hand, I see that it's already in use in several other places in moto without it being listed here. It's been imported by `moto/identitystore/models.py` since June, and by `moto/utilities/aws_headers.py` April, but there is behind the TYPE_CHECKING gate

If you want to avoid the runtime dependency, I'll fix `moto/identitystore/models.py` too.